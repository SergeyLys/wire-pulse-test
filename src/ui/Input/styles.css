.input {
  outline: none;
  width: 100%;
  background-color: #fff;
  color: #000;
  padding: 10px;
  border-radius: 10px;
  border: 1px solid rgba(0, 0, 0, 0.5);
}
