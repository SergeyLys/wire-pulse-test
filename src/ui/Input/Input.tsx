import { memo, useState } from "react";
import "./styles.css";

type Props = {
  value: string;
  onChange: (value: string) => void;
  testId?: string;
}

const Input: React.FC<Props> = ({ value, onChange, testId }) => {
  const [inputValue, setInputValue] = useState<string>(value);

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(e.target.value);
    onChange(e.target.value);
  };

  return (
    <input
      data-testid={testId || "input-test-id"}
      className="input"
      type="text"
      value={inputValue}
      onChange={handleChange}
    />
  );
};

export type {Props};
export default memo(Input);
