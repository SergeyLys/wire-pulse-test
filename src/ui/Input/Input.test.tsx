import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Input, {Props} from './Input';

const Component = (props: Props) => (
  <Input {...props} />
);

const setupComponent = (props?: Record<string, any>) => {
  const defaultProps = {
    value: '',
    onChange: jest.fn()
  };

  const mergedProps = {...defaultProps, ...props};

  return <Component {...mergedProps} />;
}

describe('Input component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(setupComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should display correct value if value is provided', () => {
    render(setupComponent({value: 'Test value'}))

    expect(screen.getByTestId('input-test-id')).toHaveValue('Test value');
  });

  it('should react on change function', () => {
    const props = {
      value: '',
      onChange: jest.fn()
    }

    const mockOnChange = jest.spyOn(props, 'onChange');

    render(setupComponent(props));

    fireEvent.change(screen.getByTestId('input-test-id'), {
      target: {
        value: 'some typed text'
      }
    });

    expect(screen.getByTestId('input-test-id')).toHaveValue('some typed text');
    expect(mockOnChange).toHaveBeenCalledWith('some typed text');
  });
});
