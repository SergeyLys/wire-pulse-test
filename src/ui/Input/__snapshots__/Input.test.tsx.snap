// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders correctly 1`] = `
<input
  className="input"
  data-testid="input-test-id"
  onChange={[Function]}
  type="text"
  value=""
/>
`;
